lib/elixir/test/elixir/kernel/overridable_test.exs

```diff
@@ -214,4 +216,20 @@ defmodule Kernel.OverridableTest do
       """
     end
   end
+
+  test "overriding private functions is deprecated and emits a warning" do
```

Comment:
With addition of this test OverridableTest needs to be async: false.
I think we can skip this test since it is a deprecation and I believe we usually skip testing them.
Otherwise we can define `Kernel.OverridableTest.Warnings` with `async: false`.